src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,106 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_OF_MAP = 0.75f;
+    private Node<K, V>[] fieldOfMap;
+    private int size;
+
+    public MyHashMap() {
+        this.fieldOfMap = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+    }
+
+    static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkCapacity();
+        int position = hash(key, fieldOfMap.length);
+        Node<K, V> node = new Node<>(key, value);
+        Node<K, V> current;
+        if (fieldOfMap[position] == null) {
+            fieldOfMap[position] = node;
+            size++;
+            return;
+        }
+        current = fieldOfMap[position];
+        do {
+            if (Objects.equals(current.key, key)) {
+                current.value = node.value;
+                return;
+            } else if (current.next == null) {
```

Comment:
`else if` useless, just `if` would be enough, since you have `return` statement in previous block.